terra-workspace-manager/sa.tf

```diff
@@ -21,7 +21,10 @@ locals {
     "roles/cloudprofiler.agent", # Profiling
     "roles/cloudtrace.agent", # Tracing for monitoring
     "roles/monitoring.editor", # Exporting metrics
-    "roles/pubsub.editor" # Creating, publishing & subscribing pub/sub topics for multi-instance Stairway.
+    "roles/pubsub.editor", # Creating, publishing & subscribing pub/sub topics for multi-instance Stairway.
+    "roles/storage.objectViewer", # read source bucket objects for storage transfer
+    "roles/storage.legacyBucketReader", # read source bucket for storage transfer
+    "roles/storage.legacyBucketWriter", # write sink bucket for storage transfer
```

Comment:
I'm not at all familiar with the transfer service, but iiuc this change grants bucket read/write permission in the control plane project to the WSM service account. That seems like it might be backwards - shouldn't it have read permission in the data plane projects, and something like `roles/storagetransfer.user` in the control plane?

I'm also seeing something about a GCP managed service account doing the actual transfer, so we may need to grant write permission to that account instead of the WSM SA (https://cloud.google.com/storage-transfer/docs/iam-transfer#sink-permissions)

Reply:
From reading the GCP page above, feels there are two thing we need to figure out:
1. Do we need to use `project-PROJECT_NUMBER@storage-transfer-service.iam.gserviceaccount.com`? If so, the problem would be harder, we need to grant the `project-source@storage-transfer-service.iam.gserviceaccount.com` permissions as https://cloud.google.com/storage-transfer/docs/iam-transfer#source-permissions described. Also similar step to the sink-permission stuff. And those IAM granting can not be done in terraform, it would be two extra GCP IAM calls happens ether creating GCP context or cloning workspace. (I looked up some GCP projects, and did not find those SA account even exists today).
2. Can we specify our custom SA accounts? If so, the problem simpler. Because WSM-SA is already the owner of those GCP projects. So as long as you set the correct SA, it would just work.

Reply:
@zloery I think you're right and that I interpreted that doc wrong. It makes more sense to have the read permissions in the source project (or just on the bucket?) and likewise for the sink project.